src/vm/hints/sha256_utils.rs

```diff
@@ -0,0 +1,242 @@
+use crate::{
+    bigint,
+    serde::deserialize_program::ApTracking,
+    vm::{
+        errors::vm_errors::VirtualMachineError,
+        hints::hint_utils::{
+            bigint_to_u32, get_integer_from_var_name, get_ptr_from_var_name,
+            get_relocatable_from_var_name, insert_value_from_var_name,
+        },
+        vm_core::VirtualMachine,
+    },
+};
+
+use generic_array::GenericArray;
+use num_bigint::BigInt;
+use num_traits::{One, Zero};
+use sha2::compress256;
+use std::collections::HashMap;
+
+const SHA256_INPUT_CHUNK_SIZE_FELTS: usize = 16;
+const SHA256_STATE_SIZE_FELTS: usize = 8;
+const BLOCK_SIZE: usize = 7;
+
+pub fn sha256_input(
+    vm: &mut VirtualMachine,
+    ids: &HashMap<String, BigInt>,
+    hint_ap_tracking: Option<&ApTracking>,
+) -> Result<(), VirtualMachineError> {
+    let n_bytes = get_integer_from_var_name(
+        "n_bytes",
+        ids,
+        &vm.memory,
+        &vm.references,
+        &vm.run_context,
+        hint_ap_tracking,
+    )?;
+
+    insert_value_from_var_name(
+        "full_word",
+        if n_bytes >= &bigint!(4) {
+            BigInt::one()
+        } else {
+            BigInt::zero()
+        },
+        ids,
+        &mut vm.memory,
+        &vm.references,
+        &vm.run_context,
+        hint_ap_tracking,
+    )
+}
+
+pub fn sha256_main(
+    vm: &mut VirtualMachine,
+    ids: &HashMap<String, BigInt>,
+    hint_ap_tracking: Option<&ApTracking>,
+) -> Result<(), VirtualMachineError> {
+    let sha256_start = get_relocatable_from_var_name(
+        "sha256_start",
+        ids,
+        &vm.memory,
+        &vm.references,
+        &vm.run_context,
+        hint_ap_tracking,
+    )?;
+
+    let input_ptr = vm.memory.get_relocatable(&sha256_start)?;
```

Comment:
Couldn't this be changed to `let input_ptr = get_ptr_from_var_name("sha256_start" ...)`

Reply:
Resolved: https://github.com/lambdaclass/cleopatra_cairo/pull/357/commits/65d94f9227e3e5caee8fd4e1738407c7fd44b523